src/lib/mongoBackend/MongoGlobal.cpp

```diff
@@ -1110,11 +1110,40 @@ bool entitiesQuery
     }
 
     /* Process query result */
-    while (cursor->more()) {
+    while (cursor->more())
+    {
+        BSONObj                 r    = cursor->next();
+        std::string             err  = r.getStringField("$err");
+        ContextElementResponse* cer  = new ContextElementResponse();
+
+        if (err != "")
+        {
+          //
+          // We can't return the error 'as is', as it may contain forbidden characters.
```

Comment:
What do you mean by "failed"? That the forbidden characteres didn't get escaped?